server/fleet/users.go

```diff
@@ -24,7 +24,12 @@ type User struct {
 	APIOnly    bool    `json:"api_only" db:"api_only"`
 
 	// Teams is the teams this user has roles in.
+	// For users with a global role, Teams is expected to be empty.
 	Teams []UserTeam `json:"teams"`
+
+	// For users with a global role, AvailableTeams includes all teams for the organization.
+	// For non-global users, AvailableTeams includes only the teams where the user has been assigned a role.
+	AvailableTeams []Team `json:"available_teams"`
```

Comment:
I don't think `AvailableTeams` should be a part of the `User` struct, but instead added alongside the user in the response structs for the needed APIs.

I would only add to `User` what is a property of a user. I don't think `AvailableTeams` is a property of them, but rather what they have the right to be aware of, while in contrast `Teams` are the teams the user belongs to.

So I would add it to `getUserResponse` and `loginResponse`, so that you would get:

```
{
  user: {...},
  available_teams: [...]
}
```

What do you think?